Source/Scene/ModelExperimental/InstancingPipelineStage.js

```diff
@@ -423,15 +764,107 @@ function processMatrixAttributes(node, count, renderResources, frameState) {
   ];
 
   const shaderBuilder = renderResources.shaderBuilder;
-  shaderBuilder.addDefine("HAS_INSTANCE_MATRICES");
-  shaderBuilder.addAttribute("vec4", "a_instancingTransformRow0");
-  shaderBuilder.addAttribute("vec4", "a_instancingTransformRow1");
-  shaderBuilder.addAttribute("vec4", "a_instancingTransformRow2");
+  shaderBuilder.addAttribute("vec4", `a_instancing${attributeString}Row0`);
+  shaderBuilder.addAttribute("vec4", `a_instancing${attributeString}Row1`);
+  shaderBuilder.addAttribute("vec4", `a_instancing${attributeString}Row2`);
+
+  instancingVertexAttributes.push.apply(
+    instancingVertexAttributes,
+    matrixAttributes
+  );
+}
 
-  return instancingVertexAttributes;
+function processVec3Attribute(
+  renderResources,
+  attribute,
+  typedArray,
+  frameState,
+  instancingVertexAttributes,
+  attributeString
+) {
+  let buffer = attribute.buffer;
+  let byteOffset = attribute.byteOffset;
+  let byteStride = attribute.byteStride;
+  if (!defined(buffer)) {
+    buffer = Buffer.createVertexBuffer({
+      context: frameState.context,
+      typedArray: typedArray,
+      usage: BufferUsage.STATIC_DRAW,
+    });
+    byteOffset = 0;
+    byteStride = undefined;
+  }
+
+  instancingVertexAttributes.push({
+    index: renderResources.attributeIndex++,
+    vertexBuffer: buffer,
+    componentsPerAttribute: 3,
+    componentDatatype: ComponentDatatype.FLOAT,
+    normalize: false,
+    offsetInBytes: byteOffset,
+    strideInBytes: byteStride,
+    instanceDivisor: 1,
+  });
+
+  const shaderBuilder = renderResources.shaderBuilder;
+  shaderBuilder.addAttribute("vec3", `a_instance${attributeString}`);
+}
+
+function processFeatureIdAttributes(
+  renderResources,
+  frameState,
+  instances,
+  instancingVertexAttributes,
+  transformMatrices
+) {
+  const attributes = instances.attributes;
+  const model = renderResources.model;
+  const shaderBuilder = renderResources.shaderBuilder;
+
+  // Load Feature ID vertex attributes. These are loaded as typed arrays in GltfLoader
+  // because we want to expose the instance feature ID when picking.
+  for (let i = 0; i < attributes.length; i++) {
+    const attribute = attributes[i];
+    if (attribute.semantic !== InstanceAttributeSemantic.FEATURE_ID) {
+      continue;
+    }
+
+    if (
+      attribute.setIndex >= renderResources.featureIdVertexAttributeSetIndex
+    ) {
+      renderResources.featureIdVertexAttributeSetIndex = attribute.setIndex + 1;
+    }
+
+    const vertexBuffer = Buffer.createVertexBuffer({
+      context: frameState.context,
+      typedArray: attribute.packedTypedArray,
+      usage: BufferUsage.STATIC_DRAW,
+    });
+    vertexBuffer.vertexArrayDestroyable = false;
+    model._resources.push(vertexBuffer);
```

Comment:
Just in case there's a merge conflict with #10397 once that gets merged, make sure the statistics get updated here.